metricflow/test/query/test_query_parser.py

```diff
@@ -240,6 +241,19 @@ def test_parse_and_validate_where_constraint_dims(time_spine_source: TimeSpineSo
     assert DimensionSpec(element_name="is_instant", identifier_links=()) not in query_spec.dimension_specs
 
 
+def test_parse_and_validate_where_constraint_metric_time(time_spine_source: TimeSpineSource) -> None:
+    """Test that granularity of metric_time reference in where constraint is at least that of the ds dimension."""
+    revenue_yaml_file = YamlConfigFile(filepath="inline_for_test_2", contents=REVENUE_YAML)
+
+    query_parser = query_parser_from_yaml([revenue_yaml_file], time_spine_source)
+    with pytest.raises(RequestTimeGranularityException):
```

Comment:
You might want to add the `match` param here to assert that the correct error message is being thrown, but given that this is a granularity-specific exception I don' think it's necessary.